crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -842,7 +844,7 @@ impl<'a> TimelineEventHandler<'a> {
                     // In more complex cases, remove the item and day
                     // divider (if necessary) before re-adding the item.
                     trace!("Removing local echo or duplicate timeline item");
-                    self.items.remove(idx);
+                    removed_event_item_id = Some(self.items.remove(idx).internal_id);
```

Comment:
It's a function-local `Option` so I'll choose to ignore this for now. Ideally this would be a conditionally-initialized variable rather than an `Option`, but there's no way of conditionally reading those if initialized (even though the compiler has to track initialized-ness for conditional dropping anyways)..